internal/nostr/expo.go

```diff
@@ -0,0 +1,164 @@
+package nostr
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/nbd-wtf/go-nostr"
+	expo "github.com/oliveroneill/exponent-server-sdk-golang/sdk"
+	"github.com/sirupsen/logrus"
+)
+
+func (svc *Service) NIP47ExpoNotificationHandler(c echo.Context) error {
+	var requestData NIP47ExpoNotificationRequest
+	if err := c.Bind(&requestData); err != nil {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "Error decoding notification request",
+			Error:   err.Error(),
+		})
+	}
+
+	if (requestData.PushToken == "") {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "push token is empty",
+			Error:   "no push token in request data",
+		})
+	}
+
+	_, err := expo.NewExponentPushToken(requestData.PushToken)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "invalid push token",
+			Error:   "invalid push token in request data",
+		})
+	}
+
+	if (requestData.WalletPubkey == "") {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "wallet pubkey is empty",
+			Error:   "no wallet pubkey in request data",
+		})
+	}
+
+	if (requestData.ConnPubkey == "") {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "connection pubkey is empty",
+			Error:   "no connection pubkey in request data",
+		})
+	}
+
+	var existingSubscriptions []Subscription
+	if err := svc.db.Where("push_token = ? AND open = ?", requestData.PushToken, true).Find(&existingSubscriptions).Error; err != nil {
+		svc.Logger.WithError(err).WithFields(logrus.Fields{
+			"push_token": requestData.PushToken,
+		}).Error("Failed to check existing subscriptions")
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "internal server error",
+			Error:   err.Error(),
+		})
+	}
+
+	for _, existingSubscription := range existingSubscriptions {
```

Comment:
could the DB query above be improved instead of having to do a for loop after (more filtering?)